### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by CoderOfTomorrow**
//startedCashByCode option
//small changes everywhere

### CODE REVIEW

File: Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            return CommandResult.ReturnSuccess();
+        }
+
+
+        private async Task LifeSpan(Guid Id,IServiceScopeFactory serviceScopeFactory)
+        {
+            await Task.Delay(3600000);
+            using var scope = serviceScopeFactory.CreateScope();
+            var dbContext = scope.ServiceProvider.GetService<ApplicationDbContext>();
+            var itemToRemove = dbContext.CashByCodeRepo.FirstOrDefault(e => e.Id == Id);
```
Comment (TermanEmil) [inline Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs:71]: what if itemToRemove can't be found?

File: Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            context.Add(cashByCode);
+            context.SaveChanges();
+
+            _ = Task.Run(() => LifeSpan(cashByCode.Id, serviceScopeFactory));
+
+            return CommandResult.ReturnSuccess();
+        }
+
+
+        private async Task LifeSpan(Guid Id,IServiceScopeFactory serviceScopeFactory)
```
Comment (TermanEmil) [inline Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs:66]: How about we trigger an event when a CashByCode is created?
```csharp
public class CashByCodeCreated : INotification
...
mediator.Publish(new CashByCodeCreated(...));
...
public class CashByCodeCreatedHandler : INotificationHandler<CashByCodeCreated>()
```

something along these lines.

In this way, the logic of creating a cash by code doesn't have unnecessary responsibilities (like deleting it after X seconds). (Single Responsibility Principle)
Also, in this way, we can attach additional  logic when a cash by code is created without modifying this class (like sending an email notification if we really want). (Open-closed Principle)

Comment (TermanEmil) [reply Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs:66]: but if this feels too complicated, we can always implement this in another PR

File: Server/Controllers/CashByCodeController.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        public async Task<List<CashByCodeDto>> GetAllCashByCode()
+        {
+            var query = new GetAllCashByCodeQuery
+            {
+                UserId = userManager.GetUserId(User)
+            };
+
+            var cashByCodeList = await mediator.Send(query);
+            var list = new List<CashByCodeDto>();
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<CashByCode, CashByCodeDto>());
```
Comment (TermanEmil) [inline Server/Controllers/CashByCodeController.cs:38]: As a side note, it's possible to create a mapping [Profile](https://docs.automapper.org/en/stable/Configuration.html). And [register](https://docs.automapper.org/en/stable/Dependency-injection.html) this profile in the Dependency Injection container.
Therefore, your controller will simply receive an IMapper and you won't need to manually configure it in each action.

File: Server/Controllers/TransactionController.cs [hunk 4]
```diff
@@ -1,81 +1,49 @@
...
-                    _ => transactionsList.OrderByDescending(e => e.Date).ToList()
-                },
-
-
-                "Currency" => transactionsList = orderFilter switch
-                {
-                    "Ascendent" => transactionsList.OrderBy(e => e.Currency).ThenBy(e => e.Amount).ToList(),
-                    _ => transactionsList.OrderByDescending(e => e.Currency).ThenByDescending(e => e.Amount).ToList()
-                },
+            var transactionsList = mediator.Send(query).GetAwaiter().GetResult();
```
Comment (TermanEmil) [inline Server/Controllers/TransactionController.cs:45]: It's generally not a good idea to do this. `GetAwaiter().GetResult()` will block the thread until the task is finished.
You could simply make this controller action async. Why not?

File: Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs (hunk 1 above)
Comment (CoderOfTomorrow) [reply Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs:71]:
```
> private async Task LifeSpan(Guid Id,IServiceScopeFactory serviceScopeFactory)
        {
            await Task.Delay(100000);
            using var scope = serviceScopeFactory.CreateScope();
            var dbContext = scope.ServiceProvider.GetService<ApplicationDbContext>();
            if (dbContext.CashByCodeRepo.Any(e => e.Id == Id))
            {
                var itemToRemove = dbContext.CashByCodeRepo.FirstOrDefault(e => e.Id == Id);
                var wallet =  await dbContext.Users.
                    Where(u => u.Id == itemToRemove.SourceUserId).
                    SelectMany(u => u.Wallets).
                    Where(w => w.Id == itemToRemove.SourceWalletId).
                    FirstOrDefaultAsync();
                wallet.Amount += itemToRemove.Amount;
                dbContext.CashByCodeRepo.Remove(itemToRemove);
                await dbContext.SaveChangesAsync();
            }
        }
```

this is the code right now

File: Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs (hunk 2 above)
Comment (CoderOfTomorrow) [reply Server/Application/CashByCodeMethods/Command/GenerateCashByCodeCommand.cs:66]: complicaterd or not , if i want to become a good .net programmator i need to explore and learn a lot of things .

File: Server/Controllers/TransactionController.cs (hunk 4 above)
Comment (CoderOfTomorrow) [reply Server/Controllers/TransactionController.cs:45]: I changed it . Thx

**Pull request merged**